### PR TITLE
Include troubleshooting guide link in exception messages

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/30692

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs [hunk 1]
```diff
@@ -49,5 +49,12 @@ internal static class Constants
         public const int WellKnownPublicPortsLimit = 1023;
 
         public const string DefaultScope = "https://servicebus.azure.net/.default";
+
+        /// <summary>
+        /// The message appended to exceptions returned from the service that contains a link to the troubleshooting guide.
+        /// Usage errors with obvious causes do not contain this message.
+        /// </summary>
+        internal const string TroubleshootingMessage =
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs:57]: Is there a reason this is `internal` and everything else in the class is `public`?

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusException.cs [hunk 2]
```diff
@@ -46,7 +42,6 @@ public class ServiceBusException : Exception
         /// <summary>
         ///   Gets a message that describes the current exception.
         /// </summary>
-        ///
         public override string Message
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusException.cs:45]: This is flipping brilliant and I'm stealing it!

**Review by jsquire — APPROVED** [review]
image

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Constants.cs:57]: Nope! will update for consistency.

**Pull request merged**